framework/src/problems/FEProblemBase.C

```diff
@@ -3332,64 +3328,86 @@ FEProblemBase::projectSolution()
+    Threads::parallel_reduce(bnd_nodes, cbic);
 
-  // Also, load values into the SCALAR dofs
-  // Note: We assume that all SCALAR dofs are on the
-  // processor with highest ID
-  if (processor_id() == (n_processors() - 1) && _scalar_ics.hasActiveObjects())
-  {
-    const auto & ics = _scalar_ics.getActiveObjects();
-    for (const auto & ic : ics)
+    // close the solution vectors to make sure the solutions are in good state before
+    // copyOldSolutions()
+    for (auto & nl : _nl)
+      nl->solution().close();
+    _aux->solution().close();
+
+    // Also, load values into the SCALAR dofs
+    // Note: We assume that all SCALAR dofs are on the
+    // processor with highest ID
+    if (processor_id() == (n_processors() - 1) && _scalar_ics.hasActiveObjects())
     {
-      MooseVariableScalar & var = ic->variable();
-      var.reinit();
+      const auto & ics = _scalar_ics.getActiveObjects();
+      for (const auto & ic : ics)
+      {
+        MooseVariableScalar & var = ic->variable();
+        var.reinit();
 
-      DenseVector<Number> vals(var.order());
-      ic->compute(vals);
+        DenseVector<Number> vals(var.order());
+        ic->compute(vals);
 
-      const unsigned int n_SCALAR_dofs = var.dofIndices().size();
-      for (unsigned int i = 0; i < n_SCALAR_dofs; i++)
-      {
-        const dof_id_type global_index = var.dofIndices()[i];
-        var.sys().solution().set(global_index, vals(i));
-        var.setValue(i, vals(i));
+        const unsigned int n_SCALAR_dofs = var.dofIndices().size();
+        for (unsigned int i = 0; i < n_SCALAR_dofs; i++)
+        {
+          const dof_id_type global_index = var.dofIndices()[i];
+          var.sys().solution().set(global_index, vals(i));
+          var.setValue(i, vals(i));
+        }
       }
     }
-  }
 
-  for (auto & sys : _solver_systems)
-  {
-    sys->solution().close();
-    sys->solution().localize(*sys->system().current_local_solution, sys->dofMap().get_send_list());
+    for (auto & sys : _solver_systems)
+    {
+      sys->solution().close();
+      sys->solution().localize(*sys->system().current_local_solution,
+                               sys->dofMap().get_send_list());
+    }
+
+    _aux->solution().close();
+    _aux->solution().localize(*_aux->sys().current_local_solution, _aux->dofMap().get_send_list());
+
+    // calling copyOldSolutions() to send ICs for old and older states back to where they
+    // should be before the start of the time-looping.
+    // Functionality moved here from FEProblemBase::initialSetup().
+    if (_global_current_state > 0)
+    {
+      for (auto & sys : _solver_systems)
+        sys->copySolutionsBackwards();
+      _aux->copySolutionsBackwards();
+    }
   }
 
-  _aux->solution().close();
-  _aux->solution().localize(*_aux->sys().current_local_solution, _aux->dofMap().get_send_list());
+  // Resetting _global_current_state to 0 in case anyone needs to run ICs again (which they do
+  // sometimes).
```

Comment:
In the comment, can you spell out the situations in which people want to run ICs again?